pakyow-core/lib/core/middleware/req_path_normalizer.rb

```diff
@@ -0,0 +1,24 @@
+module Pakyow
+  module Middleware
+    # Rack compatible middleware that normalize the path if contains '//',
+    # it replace '//' with '/' and issue a 301 redirect to the new path.
+    #
+    # @api public
+    class ReqPathNormalizer
+      def initialize(app)
+        @app = app
+      end
+
+      def call(env)
+        if env['PATH_INFO'].include? '//'
+          # normalize the path and issue 301 redirect
+          env['PATH_INFO'].gsub!('//', '/')
```

Comment:
Fwiw, I'd probably just pull the part you're interested in from the ENV rather than dup'ing the whole thing. Maybe something like

```ruby
request_path = env["PATH_INFO"]
normalized_path = request_path.gsub("//", "/")
```

I'm not honestly not sure how dup works exactly, but I imagine this might be more efficient.